staging/src/k8s.io/apiserver/pkg/endpoints/discovery/version.go

```diff
@@ -78,6 +80,57 @@ func (s *APIVersionHandler) handle(req *restful.Request, resp *restful.Response)
 }
 
 func (s *APIVersionHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
-	responsewriters.WriteObjectNegotiated(s.serializer, negotiation.DefaultEndpointRestrictions, schema.GroupVersion{}, w, req, http.StatusOK,
-		&metav1.APIResourceList{GroupVersion: s.groupVersion.String(), APIResources: s.apiResourceLister.ListAPIResources()})
+	reqURL := req.URL
+	if reqURL == nil {
+		// Can not find documentation guaranteeing the non-nility of reqURL.
+		w.WriteHeader(500)
+		return
+	}
+
+	// Get current resource list and hash
+	resourceList, hash := s.apiResourceLister.ListAPIResources()
```

Comment:
Why don't we hash that here, as opposed to receiving the hash?

Reply:
The biggest problem with that is how we're going to route the hash back to servergroups though.

Reply:
It would be impossible/expensive to know if a re-hash is necessary

Reply:
because we would have to serialize every time?

Reply:
Yes, if there was a way to know if the hash/object has changed without reserializing it or deep traversing it, then this would not be necessary.